python/lsst/meas/extensions/scarlet/scarletDeblendTask.py

```diff
@@ -123,7 +123,7 @@ def _computePsfImage(self, position=None):
     for bidx, single in enumerate(self.singles):
         try:
             if position is None:
-                psf = single.getPsf().computeImage()
+                psf = single.getPsf().computeImage(single.getPsf().getAveragePosition())
```

Comment:
Paging @fred3m

Reply:
Honestly, it should probably return an error, since using the average position can cause unexpected problems.

Reply:
Thinking back on this as to why that behavior is coded in the first place, `computeKernelImage` _definitely_ didn't work properly if no position is given. At the time, `computeKernel` didn't have the same bug, so that's why we still allowed it. But I do think that raising an Exception if no position is specified is the correct behavior now.

Reply:
In that case, I can close this PR without merging and perhaps @fred3m can create another ticket to raise an exception?

Reply:
I would do both.  First add the explicit position, file a ticket on @fred3m, and add a TODO referencing the ticket.

Reply:
@erykoff  I disagree. This code has not worked since calling `computeImage` with no position was deprecated, and whoever did that missed removing it here. So I agree that @arunkannawadi should either change this code to raise an Exception or assign a ticket to me to do that. Deprecating that behavior was done for a good reason and I don't want it to slip back in.

Reply:
When computeImage was deprecated _nothing_ was removed.  So this is kind of cleaning that up.  The idea is not that it doesn't work, we want to explicitly use the average position.  More after my meeting.

Reply:
DM-34770 ticket touches multiple packages and all it does to use the implicit default position explicitly when accurate or switch to source/bbox center locations where appropriate. Making position required here and throwing an exception if None should certainly not be as part of this ticket. I am okay either way as long as the newly filed DM-34777 is merged quickly.

Reply:
Ok, we should discuss this. I have a presentation somewhere from a few years ago that describes why this is a bad idea. So even if it is what you need for your use case, I don't want to make this the default behavior because unless someone knows that they're doing it's easy to make a mistake that can have catastrophic consequences.

Reply:
I think this branch is actually unused, which is good.  So I think that DM-34777 is trivial.

Reply:
Sorry @arunkannawadi , it looks like our comments crossed and I didn't see yours before I submitted. So looking at that ticket I understand why this is being updated now. However, I stand by my original assertion that this function should never take `position=None` as an argument. Currently if one tries to use `position=None` in the stack this will display a future warning (I just tried with the latest weekly). I would prefer that to not notifying the user that their use of `position=None` is a bad idea. So if you don't want to raise an exception in this ticket please revert this change, as I prefer the current behavior that will at least notify people that they are doing something bad (or wait for DM-37777).

Reply:
Okay, I think the thing to do is not make this change, and add a TODO DM-37777 which hopefully is done soon.

Reply:
I thought the point of this ticket was to make it so that the future warning was handled? If `position=None` is not allowed then raise, don't wait for the warning.

Reply:
I did just finish DM-34777 and if Arun prefers, he can just merge it into this ticket branch instead of main. As @erykoff said, the change was trivial.